src/libraries/System.Security.Cryptography/tests/X509Certificates/X500DistinguishedNameBuilderTests.cs

```diff
@@ -0,0 +1,379 @@
+        {
+            AssertBuilder(
+                "30223120301E06092A864886F70D01090116116B6576696E406578616D706C652E636F6D",
+                builder => builder.AddEmailAddress("kevin@example.com"));
+        }
+
+        [Fact]
+        public static void AddEmailAddress_ExceedsMaximumLength_Fails()
+        {
+            X500DistinguishedNameBuilder builder = new();
+            string bigEmail = new string('a', 244) + "@example.com";
+            AssertExtensions.Throws<ArgumentException>("emailAddress", () => builder.AddEmailAddress(bigEmail));
+        }
+
+        [Fact]
+        public static void AddEmailAddress_InvalidIA5_Fails()
+        {
+            X500DistinguishedNameBuilder builder = new();
+            ArgumentException ex = AssertExtensions.Throws<ArgumentException>(
+                "emailAddress",
+                () => builder.AddEmailAddress("\u043A@example.com"));
+            Assert.Contains("'IA5String'", ex.Message);
+        }
+
+        [Fact]
+        public static void AddEmailAddress_NullOrEmpty_Fails()
+        {
+            AssertAddThrowsOnNullAndEmpty("emailAddress", builder => builder.AddEmailAddress);
+        }
+
+        [Fact]
+        public static void AddCommonName_Success_EncodeToUTF8String()
+        {
+            AssertBuilder("300D310B300906035504030C024B4A", builder => builder.AddCommonName("KJ"));
+        }
+
+        [Fact]
+        public static void AddCommonName_NullOrEmpty_Fails()
+        {
+            AssertAddThrowsOnNullAndEmpty("commonName", builder => builder.AddCommonName);
+        }
+
+        [Fact]
+        public static void AddLocalityName_Success_EncodeToUTF8String()
+        {
+            AssertBuilder("300F310D300B06035504070C04486F6D65", builder => builder.AddLocalityName("Home"));
+        }
+
+        [Fact]
+        public static void AddLocalityName_NullOrEmpty_Fails()
+        {
+            AssertAddThrowsOnNullAndEmpty("localityName", builder => builder.AddLocalityName);
+        }
+
+        [Fact]
+        public static void AddCountryOrRegion_Success_EncodeToPrintableString()
+        {
+            AssertBuilder("300D310B3009060355040613025553", builder => builder.AddCountryOrRegion("US"));
+        }
+
+        [Theory]
+        [InlineData("A")]
+        [InlineData("AAA")]
+        public static void AddCountryOrRegion_InvalidLength_Fails(string countryOrRegion)
+        {
+            X500DistinguishedNameBuilder builder = new();
+            AssertExtensions.Throws<ArgumentException>(
+                "twoLetterCode",
+                () => builder.AddCountryOrRegion(countryOrRegion));
+        }
+
+        [Fact]
+        public static void AddCountryOrRegion_InvalidContents_Fails()
+        {
+            X500DistinguishedNameBuilder builder = new();
+            ArgumentException ex = AssertExtensions.Throws<ArgumentException>(
+                "twoLetterCode",
+                () => builder.AddCountryOrRegion("``"));
```

Comment:
The comments called out a surrogate pair as an interesting case.  May as well throw one in for the tests.